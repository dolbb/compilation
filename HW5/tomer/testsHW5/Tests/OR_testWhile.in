int Int(){print("set!\n");return 500;}
void WasHere(){print("was here\n");}
void main(){
int target = 0;
while(target == 0) target = Int();
printi(target);
print("\n");target = 0;
while(true){
int t6 = Int();
printi(t6);
break;
}
print("\n");while(true){
int t6;
t6 = Int();
printi(t6);
break;
}
print("\n");while(target == 0) {target = Int();}
printi(target);
print("\n");
target = 0;
while(true) {WasHere();break;}
while(true) if (true) {WasHere();break;}
while(true) if (true) {WasHere();break;} 
	else int x;
while(true)  if (false) int x; 
	else {WasHere();break;}
while(true) {
	while (true) {WasHere();break;}
break;}
while(true) {
switch (target) {
case 0: WasHere();break;
case 1b: int x;break;
};
break;}
while(target==0) while (target == 0) target = Int();
}
