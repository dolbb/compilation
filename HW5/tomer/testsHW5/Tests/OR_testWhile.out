Loaded: ./exceptions.s
set!
500
set!
500
set!
500
set!
500
was here
was here
was here
was here
was here
was here
set!
